Content.Server/StationRecords/Systems/StationRecordsSystem.cs

```diff
@@ -95,6 +120,44 @@ private void CreateGeneralRecord(EntityUid station, EntityUid player, HumanoidCh
         TryComp<DnaComponent>(player, out var dnaComponent);
 
         CreateGeneralRecord(station, idUid.Value, profile.Name, profile.Age, profile.Species, profile.Gender, jobId, fingerprintComponent?.Fingerprint, dnaComponent?.DNA, profile, records);
+
+        var query = EntityQueryEnumerator<SectorStationRecordComponent>();
+
+        while (query.MoveNext(out var stationEntityUid, out var comp))
+        {
+            if (!TryComp<IgnoreSectorStationRecordComponent>(player, out var playerComp))
+            {
+                var stationList = EntityQueryEnumerator<StationRecordsComponent>();
+
+                while (stationList.MoveNext(out var stationUid, out var stationRecComp))
+                {
+                    if (TryComp<StationRecordKeyStorageComponent>(idUid.Value, out var keyStorage)
+                    && stationEntityUid != null
+                    && keyStorage.Key != null)
+                    {
+                        if (!TryGetRecord<GeneralStationRecord>(keyStorage.Key.Value, out var record))
+                            continue;
+
+                        AddRecordEntry((EntityUid) stationEntityUid, record);
+                        break;
+                    }
+                }
+
+                TryComp<StationRecordsComponent>(stationEntityUid, out var stationRec);
+
+                //Checks if certain information should be faked, is yes then will randomise it
+                string playerJob = jobId;
+                if(TryComp<FakeSectorStationRecordComponent>(player, out var playerComponent))
+                {
+                    string[] jobs = { "Contractor", "Pilot", "Mercenary" };
+                    // Randomises job
+                    var random = _robustRandom.Next(jobs.Length);
+                    playerJob = jobs[random];
+                }
+                CreateGeneralRecord(stationEntityUid, idUid.Value, profile.Name, profile.Age, profile.Species, profile.Gender, playerJob, fingerprintComponent?.Fingerprint, dnaComponent?.DNA, profile, stationRec!);
+            }
+
+        }
```

Comment:
``// Frontier`` block of code

Reply:
The changes that were added and then by Whatston3 should mark it correctly